client/profile-wizard/steps/business-details/flows/selective-bundle/index.js

```diff
@@ -153,6 +129,37 @@ class BusinessDetails extends Component {
 			} );
 	}
 
+	async pushProfileUpdates( additions = {} ) {
+		const { updateProfileItems } = this.props;
+
+		const {
+			other_platform: otherPlatform,
+			other_platform_name: otherPlatformName,
+			product_count: productCount,
+			revenue,
+			selling_venues: sellingVenues,
+		} = this.state.savedValues;
+
+		const updates = {
+			other_platform: otherPlatform,
+			other_platform_name:
+				otherPlatform === 'other' ? otherPlatformName : '',
+			product_count: productCount,
+			revenue,
+			selling_venues: sellingVenues,
+			...additions,
+		};
+
+		// Remove possible empty values like `revenue` and `other_platform`.
+		Object.keys( updates ).forEach(
```

Comment:
I get that reduce can be less readable but its often safer not to mutate. 

For example, now `updates` is a different structure than when it was defined above due to props being deleted, but we have no idea what the shape is. It's not great for the next person coming to this code. 

As a general rule I would avoid inline mutation in a for each because leads to danger. If we assign a new variable it's clearer and we don't need to use `delete`.

```
const finalUpdates = Object.entries(myObj).reduce((acc, [key,val]) => {
  if (val !== '') {
    return {
      ...acc,
      [key]: val
    }
  }

  return acc
}, {})
```

We could also easily turn the above example into a utility and re-use it. For example we have some collection manipulation stuff in this file: https://github.com/woocommerce/woocommerce-admin/blob/5d56f4254b3f4118337e6f4b011960a3f0cb3e20/client/lib/collections/index.js?plain=1#L29

The other upside of that is its much easier to test there that our code does what's expected, than testing it with React present.

Reply:
I agree that in general we should avoid mutation.

Reply:
Thanks @samueljseay , to be clear this code was preexisting this PR, but I agree. Just updated 👍🏻